src/Mod/Draft/draftmake/make_polararray.py

```diff
@@ -60,6 +61,29 @@ def make_polar_array(base_object,
         It defaults to the origin `App.Vector(0, 0, 0)`.
         The vector indicating the center of rotation of the array.
 
+    axis_object: str or DocumentObject, optional
+        It defaults to `None`.
+        This parameter should be the name of an `DocumentObject` or
+        the `DocumentObject` itself.
+        If it is set the resulting array will use the referenced axis
+        with it's name provided by parameter `edge_name` that is part
+        of `axis_object` to calculate center and direction instead
```

Comment:
You call it `edge_name` but the parameter is called `axis_edge` so watch out for inconsistencies.